uitools/import/Esri/ArcGISRuntime/Toolkit/Callout.qml

```diff
@@ -209,143 +161,152 @@ Item {
+    property alias calloutHeight: root.implicitHeight
 
     /*!
-        \brief The height of the Callout.
+        \obsolete
+        Use \c{background.border.color} instead.
     */
-    property real calloutHeight: 45
+    property color borderColor
+    onBorderColorChanged: {
+        background.border.color = borderColor;
+        internal.leaderColor = borderColor;
+    }
 
-    // internal properties
-    /*! \internal */
-    property int padding: 3
-    /*! \internal */
-    property real anchorPointx: 0
-    /*! \internal */
-    property real anchorPointy: 0
-    /*! \internal */
-    property point screenCoordinates
-    /*! \internal */
-    property var adjustedLeaderPosition: Enums.LeaderPosition.Bottom
-    /*! \internal */
-    property bool calloutVisible
-    /*! \internal */
-    property real calloutMaxHeight: 45
-    /*! \internal */
-    property real calloutMinWidth: 95
-    /*! \internal */
-    property real calloutMinHeight: calloutMaxHeight
-    /*! \internal */
-    property real rectWidth: 0
-    /*! \internal */
-    property real rectHeight: 0
-    /*! \internal */
-    property real edgeBuffer: 10
-    /*! \internal */
-    property real calloutFramePadding: (2 * internalCornerRadius)
-    /*! \internal */
-    property string platform: Qt.platform.os
-    /*! \internal */
-    property real halfRectWidth: rectWidth / 2
-    /*! \internal */
-    property real halfRectHeight: rectHeight / 2
-    /*! \internal */
-    property real halfLeaderWidth: leaderWidth / 2
-    /*! \internal */
-    property real imageWidth: rectWidth / 4
-    /*! \internal */
-    property real titleWidth: Math.max(0, rectWidth - 95)
-    /*! \internal */
-    property real detailWidth: Math.max(0, rectWidth - 95)
-    /*! \internal */
-    property real cornerOffset: 15
-    /*! \internal */
-    property bool debug: false
-    /*! \internal */
-    property int internalCornerRadius: cornerRadius < 0 ? 0 : cornerRadius
-    /*! \internal */
-    property int adjustedMaxWidth: maxWidth
-    /*! \internal */
-    property int adjustedX: 0
-    /*! \internal */
-    property int adjustedY: 0
-    /*! \internal */
-    visible: false
+    /*!
+        \obsolete
+        Use \c{Callout.LeaderPosition} instead.
+    */
+    property var leaderPositionEnum: { return {
+            UpperLeft: 0,
+            Top: 1,
+            UpperRight: 2,
+            Right: 3,
+            LowerRight: 4,
+            Bottom: 5,
+            LowerLeft: 6,
+            Left: 7,
+            Automatic: 8
+        } }
 
-    /*! \internal */
-    Connections {
-        id: calloutConnection
-        target: calloutData
-        function onScreenPointChanged() {
-            anchorPointx = calloutData.screenPoint.x;
-            anchorPointy = calloutData.screenPoint.y;
+    /*!
+        \obsolete
+        Use \c{background.border.width} instead.
+    */
+    property int borderWidth
+    onBorderWidthChanged: background.border.width = borderWidth;
```

Comment:
❓ can you use binding instead of assignment? There are several of these instances

Reply:
This reads as a one-way binding, where any changes to the obsolete property will change the non-obsolete property (i.e. changes to `borderWidth` will be reflected in `background.border.width`), but the same can not be said the other way round (i.e. changes to `background.border.width` will not be reflected in `borderWidth`).

For the sake of clarity: `property int borderWidth: background.border.width` will not work by itself, as the binding only exists for as long as the user doesn't set a custom value of `borderWidth`. As soon as the user gives `borderWidth` a custom value the binding disappears.

In an ideal world, `borderWidth` would be an alias of `background.border.width`, as alias acts like a two way binding, but `alias` is strict in what you can apply it to, and `background.border.width` is an example of a property which can _not_ be aliased.

Achieving a two-way binding is possible, just more verbose than I would like, effectivey every property unrolls into:

```qml
    /*!
        \obsolete
        Use \c{background.border.width} instead.
    */
    property int borderWidth: background.border.width
    Binding on borderWidth {
        value: background.border.width

    }
    Binding {
        target: background.border
        property: "width"
        value: borderWidth
    }

    /*!
        \obsolete
        Use \c{background.color} instead.
    */
    property color backgroundColor: background.color
    Binding on backgroundColor {
        value: background.color

    }
    Binding {
        target: background
        property: "color"
        value: backgroundColor
    }

```

I'm happy to update if two-way binding behaviour is desired. I had hoped that a user would use one property or the other, not both, but I can understand the desire to be foolproof with the obsolete properties.

Reply:
I'm ok with what you have.